lib/rubocop/cop/style/yoda_condition.rb

```diff
@@ -64,6 +66,8 @@ def yoda_condition?(node)
             return false if non_equality_operator?(operator)
           end
 
+          return false if noncommutative_operator?(operator)
```

Comment:
Btw, shouldn't we actually change the definition of `comparison_method?` instead? //cc @Drenmi

Reply:
Excluding `===` from `comparison_method?` introduces unexpected behaviour of  `SafeAssignment`.

I think the code below is still unsafe assignment, but if `===` isn't a comparison, cop can't detect it.

```ruby
foo = /bar/ === 'baz'
```

So, my suggestion is to keep `===` being a comparison and skip noncommutative operation in YodaCondition cop.

Reply:
FYI: Commit to exclude === from comparision: https://github.com/fujimura/rubocop/commit/d9f4fd100bc81f46fe9aa31bea917fee37935703

Reply:
I guess the case equality operator could be a comparison method or not, depending on how one frames it. I'm not super worried about semantics here. It it's useful to remove it from the list, I say we do so. 🙂 I think it could make sense, since explicit use of `#===` is already discouraged entirely in `Style/CaseEquality`.

Reply:
Indeed.